pkg/frontend/authenticate.go

```diff
@@ -584,6 +649,105 @@ var (
 	roleIdOfRoleFormat = `select role_id from mo_catalog.mo_role where role_name = "%s";`
 
 	getRoleOfUserFormat = `select r.role_id from  mo_catalog.mo_role r, mo_catalog.mo_user_grant ug where ug.role_id = r.role_id and ug.user_id = %d and r.role_name = "%s";`
+
```

Comment:
For all these queries, it better runs blazing fast ...

Is there a plan for privilege cache?   It seems to me, that we have some priv info stored in session.  Hope we are ok.

Reply:
Right now it is just the basic privilege check logic. The privilege cache may be designed next.